commons-mongo/jvm/src/test/scala/com/avsystem/commons/mongo/SomethingPlain.scala

```diff
@@ -33,7 +33,10 @@ object SomethingPlain extends HasGenCodec[SomethingPlain] {
     string <- arbitrary[String]
     boolean <- arbitrary[Boolean]
     int <- arbitrary[Int]
-    long <- arbitrary[Long]
+    long <- Gen.oneOf(
+      Gen.choose(Long.MinValue, Int.MinValue - 1L),
+      Gen.choose(Int.MaxValue + 1L, Long.MaxValue),
+    )
     timestamp <- arbitrary[JDate]
     double <- arbitrary[Double]
     binary <- Gen.buildableOf[Array[Byte], Byte](arbitrary[Byte]).map(new Bytes(_))
```

Comment:
You could add some of the decimal / big types changed within the scope of this PR.

Reply:
Yeah but this particular test assumes very specific BSON serialization which is now more complicated (same problem as with your previous comment on arbitrary `Long`s).